src/conn.rs

```diff
@@ -83,22 +84,28 @@ pub struct Conn {
     // the schema changes. #315.
 }
 
+/// Represents an in-progress, not yet committed, set of changes to the store.
+/// Call `commit` to commit your changes, or `rollback` to discard them.
+/// A transaction is held open until you do so.
+/// Your changes will be implicitly dropped along with this struct.
 pub struct InProgress<'a, 'c> {
     transaction: rusqlite::Transaction<'c>,
     mutex: &'a Mutex<Metadata>,
     generation: u64,
     partition_map: PartitionMap,
     schema: Schema,
+    last_report: Option<Rc<TxReport>>,
```

Comment:
Why not grow a `Vec<Rc<TxReport>>` or similar?

Reply:
That's a good question. I suppose simplicity: you'll only need these reports if you want to resolve a tempid, in which case you should do so immediately after transacting on the `InProgress`. If we return a sequence, then you'll need to figure out which report you need.

On the other hand, perhaps we will need this anyway when syncing (tracking parts), so this is open to change. I'll add a comment.